packages/image_picker/example/lib/main.dart

```diff
@@ -45,17 +45,19 @@ class _MyHomePageState extends State<MyHomePage> {
         _controller.removeListener(listener);
       }
       if (isVideo) {
-        _mediaFile = ImagePicker.pickVideo(source: source).then((File _file) {
-          _controller = VideoPlayerController.file(_file)
-            ..addListener(listener)
-            ..setVolume(1.0)
-            ..initialize()
-            ..setLooping(true)
-            ..play();
-          setState(() {});
+        ImagePicker.pickVideo(source: source).then((File file) {
+          if (file != null) {
+            _controller = VideoPlayerController.file(file)
+              ..addListener(listener)
+              ..setVolume(1.0)
+              ..initialize()
+              ..setLooping(true)
+              ..play();
+            setState(() {});
```

Comment:
* By convention, state changes are made within the lambda given to `setState`. Not that it matters much to runtime behavior, but because empty `setState(() {})` calls like this are very easy to overlook, forget, or place where not needed. I believe an early version of Flutter had a `markNeedsBuild()` method or similar which was replaced by `setState` for exactly this reason.
* Since `setState` is called on completion of the `pickVideo` future, the `State` we're in may have been unmounted in the meantime. We therefore need to guard the call to `setState` with a check on the `mounted` property:
   ```dart
   if (file != null && mounted) {
     setState(() {
       _controller = ...
     });
   }
   ```
   An alternative is to use a `FutureBuilder` as for images. It does the mounted check internally.